### ISSUE TITLE
Publicly export spectral-core utils functions

### ISSUE THREAD

**Issue by Amachua** [stoplightio/spectral#2260]
**User story.**

As a developper, when I am building a tool based on Spectral, then I would like to be able to use the function available in the `spectal-core` workspace in order to reproduce the internal workflow.

**Is your feature request related to a problem?**

When I tried to bump our tool dependency with Spectral, I got the following issue --> the functions `prepareResults` and the `defaultComputeResultFingerprint` are not exported anymore.

```
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './dist/utils' is not defined by "exports" in \package\node_modules\@stoplight\spectral-core\package.json
    at new NodeError (node:internal/errors:371:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:440:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:692:3)
    at resolveExports (node:internal/modules/cjs/loader:482:36)
    at Function.Module._findPath (node:internal/modules/cjs/loader:522:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
    at Function.Module._resolveFilename.sharedData.moduleResolveFilenameHook.installedValue [as _resolveFilename] (\package\node_modules\@cspotcode\source-map-support\source-map-support.js:811:30)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```

**Describe the solution you'd like**

In addition to the `./ruleset` / `.` / `ruleset/validation`, I woudl like that the `spectral-core` is updated to export the `./utils` functions.

**Comment by P0lip** [comment]
Hey @Amachua.
I'm afraid these are internal utils we wouldn't really want to expose publicly. We don't consider them as a part of public API, and exposing them would commit us to do so. What's your actual use case?
`Spectral#run` will already ["prepare results"](https://github.com/stoplightio/spectral/blob/develop/packages/core/src/spectral.ts#L84) for you, so there's no need to use them separately.

**Comment by Amachua** [comment]
Thank you @P0lip for the feedback, sorry for not providing some feedback beforehand!

For your information, we thought we needed to use this function since we aggregated the results of several sources (files) into only one result set (wanted to apply the same logic as the one you used in your engine to ensure that we do not found several time the same issue).

We have reviewed the way we prompt the results and we no longer need to aggregate the results for the all the files but file by file --> we no longer need to use this function and we are fine with the closing of this issue. :)
